src/org/parosproxy/paros/core/scanner/VariantDdnPath.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.parosproxy.paros.core.scanner;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.SessionStructure;
+
+public class VariantDdnPath implements Variant {
+
+	private static final Logger LOGGER = Logger.getLogger(VariantDdnPath.class);
+	private final List<NameValuePair> stringParam = new ArrayList<>();
+
+	@Override
+	public void setMessage(HttpMessage msg) {
+		try {
+			List<String> treePaths = Model.getSingleton().getSession().getTreePath(msg);
+			int i = 0;

```

Comment:
Could start in 1 instead of adding one when creating the `NameValuePair`, in any case a comment explaining the reason why adding one (or starting at 1) would be better.